edx_django_utils/monitoring/internal/backends.py

```diff
@@ -103,6 +118,17 @@ def set_attribute(self, key, value):
     def record_exception(self):
         self.otel_trace.get_current_span().record_exception(sys.exc_info()[1])
 
+    def initialize_celery_monitoring(self, *args, **kwargs):
+        worker_process_init = kwargs.get('worker_process_init', None)
+        if worker_process_init is not None:
+            @worker_process_init.connect(weak=False)
+            def init_celery_tracing(*args, **kwargs):
+                self.instrumentor().instrument()
+        else:
+            raise Exception(
+                "the worker_process_init celery signal must be provided for OpenTelemetry to monitor celery tasks."
```

Comment:
If OpenTelemetry needs it, then it will have to be passed to all the backends, because there will only be one call site for the `initialize_celery_monitoring` API call we expose. However, it looks like it doesn't need to be passed in at all—this is just a signal, so we could `from celery.signals import worker_process_init`.

For that matter, it might make the most sense to instead say that the _caller_ should do that, like so:

```
@celery.signals.worker_process_init.connect(weak=False)
def init_celery_tracing(*args, **kwargs):
    edx_django_utils.monitoring.initialize_celery_monitoring()
```

We could also just leave the OpenTelemetry side of this unimplemented for now, since we won't be able to give it the testing it deserves (in production).